src/pymor/vectorarrays/block.py

```diff
@@ -179,6 +179,7 @@ def __init__(self, subspaces):
         subspaces = tuple(subspaces)
         assert all([isinstance(subspace, VectorSpaceInterface) for subspace in subspaces])
         self.subspaces = subspaces
+        self.dtype = tuple([s.dtype for s in subspaces])
```

Comment:
I think this does not make any sense. `dtype` should be the `dtype` of arrays when you call things like `dot`.

Reply:
That does not exist for the `BlockVectorArray` though, does it? I can use the the type promotion construct from `dot` though.